src/enumerable.cr

```diff
@@ -984,6 +1013,30 @@ module Enumerable(T)
     max_by? &.itself
   end
 
+  # Returns an array of the maximum `k` elements, sorted descending.
+  #
+  # It compares using `<=>` so it will work for any type that supports that method.
+  #
+  # ```
+  # [7, 5, 2, 4, 9].max(3) # => [9, 7, 5]
+  # ["Eve", "Alice", "Bob", "Mallory", "Carol"].max(2) # => ["Mallory", "Eve"]
+  # ```
+  #
+  # Returns all elements sorted descending if `k` is greater than the number of
+  # elements in the source.
+  #
+  # Raises `Enumerable::ArgumentError` if `k` is negative or if any elements are
+  # not comparable.
+  def max(k : Int) : Array(T)
+    raise ArgumentError.new("negative size #{k}") if k < 0
```

Comment:
There's no need to mention the given argument. Also, please keep the message sentence-cased.

```suggestion
    raise ArgumentError.new("Can't return negative count") if count < 0
```

Reply:
Agreed. I think this wording is most commonly used in stdlib, though:
```suggestion
    raise ArgumentError.new("Count must be positive") if k < 0
```